packages/sync/src/provider.ts

```diff
@@ -1,57 +1,59 @@
 /**
  * External dependencies
  */
-// @ts-ignore
 import * as Y from 'yjs';
 
-/** @typedef {import('./types').ObjectType} ObjectType */
-/** @typedef {import('./types').ObjectID} ObjectID */
-/** @typedef {import('./types').ObjectConfig} ObjectConfig */
-/** @typedef {import('./types').CRDTDoc} CRDTDoc */
-/** @typedef {import('./types').ConnectDoc} ConnectDoc */
-/** @typedef {import('./types').SyncProvider} SyncProvider */
+/**
+ * Internal dependencies
+ */
+import type {
+	ObjectType,
+	ObjectID,
+	ObjectConfig,
+	CRDTDoc,
+	ConnectDoc,
+	SyncProvider,
+} from './types';
 
 /**
  * Create a sync provider.
  *
- * @param {ConnectDoc} connectLocal  Connect the document to a local database.
- * @param {ConnectDoc} connectRemote Connect the document to a remote sync connection.
- * @return {SyncProvider} Sync provider.
+ * @param connectLocal  Connect the document to a local database.
+ * @param connectRemote Connect the document to a remote sync connection.
+ * @return  Sync provider.
  */
-export const createSyncProvider = ( connectLocal, connectRemote ) => {
-	/**
-	 * @type {Record<string,ObjectConfig>}
-	 */
-	const config = {};
+export const createSyncProvider = (
+	connectLocal: ConnectDoc,
+	connectRemote: ConnectDoc
+): SyncProvider => {
+	const config: Record< string, ObjectConfig > = {};
 
-	/**
-	 * @type {Record<string,Record<string,()=>void>>}
-	 */
-	const listeners = {};
+	const listeners: Record< string, Record< string, () => void > > = {};
```

Comment:
We can use the handy type alias here

```suggestion
	const listeners: Record< string, Record< string, VoidFunction > > = {};
```